src/tools/qt-gui/src/undomanager.cpp

```diff
@@ -47,10 +47,16 @@ void UndoManager::createEditKeyCommand(TreeViewModel* model, int idx, QVariantLi
 	TreeViewModel* tmpModel = qvariant_cast<TreeViewModel*>(data.takeAt(2));
 	QVariantMap oldMDMap;
 
-	foreach (ConfigNodePtr node, tmpModel->model())
+	if(tmpModel)
 	{
-		oldMDMap.insert(node->getName(), node->getValue());
+
+		foreach (ConfigNodePtr node, tmpModel->model())
+		{
+			oldMDMap.insert(node->getName(), node->getValue());
+		}
 	}
+	else
+		tmpModel = new TreeViewModel;
 
 	data.insert(2, oldMDMap);
```

Comment:
You complained that the signature of the command was too long, so I put the parameters into a map. Of course the index needs stay the same if a parameter is edited. I will add a comment that maps the indices to the items (or return to the long signature?).